### PR TITLE
API: Deprecate signal_names; rename it to component_names.

### LINKED ISSUE #448: API: signal_names -> component_names
We named `signal_names` very early in the development process -- before we had established the present distinction between `Device` and `Signal`. For clarity, I think we should rename this. We can keep the old one (with a warning).

### PR DESCRIPTION
**Pull request by danielballan**
Closes #448

### CODE REVIEW

File: doc/source/device-overview.rst [hunk 1]
```diff
@@ -117,7 +117,7 @@ and static information about the object
  ===========================  ========================================================
  :attr:`prefix`               'base' of PV name, used when building components
  ---------------------------  --------------------------------------------------------
- :attr:`signal_names`         List of the names components on this device.
+ :attr:`component_names`      List of the names components on this device.
```
Comment (tacaswell) [inline doc/source/device-overview.rst]: I am surprised rst is not choking on the indent issue on the next line...

**Comment by tacaswell** [comment]
This is approximately the usage on the floor:

```
(bleeding) ✔ ~/source/bnl/beamline_source 
10:17 $ ack signal_names
CSX/xf23id1_profiles/profile_collection/startup/20-detectors.py_preLee
50:for sig in sclr.channels.signal_names:

CSX/xf23id1_profiles/profile_collection/startup/20-detectors.py
151:for sig in sclr.channels.signal_names:

CSX/xf23id1_profiles/profile_collection/startup/Old/.20-detectors_recovered.py
35:for sig in sclr.channels.signal_names:

CSX/xf23id1_profiles/profile_collection/startup/Old/20-detectors.py~2
47:for sig in sclr.channels.signal_names:

HXN/hxntools/hxntools/device.py
19:    if not hasattr(dev, 'signal_names'):
26:    for attribute in dev.signal_names:
46:    for attribute in dev.signal_names:

HXN/hxntools/hxntools/detectors/zebra.py
15:    return [sig_name for sig_name in cls.signal_names

HXN/hxntools/hxntools/detectors/xspress3.py
82:        self.channels = list(range(1, len([_ for _ in det.signal_names

XPD/ipython_ophyd/archived/profile_collection-dan/startup/20-scalers.py
8:for ch_name in em.channels.signal_names:
15:for ch_name in sc.channels.signal_names:

XPD/ipython_ophyd/archived/profile_collection-dev/startup/20-scalers.py
8:for ch_name in em.channels.signal_names:
15:for ch_name in sc.channels.signal_names:

XPD/ipython_ophyd/profile_collection/startup/20-scalers.py
8:for ch_name in em.channels.signal_names:
15:for ch_name in sc.channels.signal_names:

SRX/ipython_ophyd/profile_xf05id1-noX11/startup/00-startup.py
24:    for chld in dev.signal_names:

SRX/ipython_ophyd/profile_xf05id1/startup/00-startup.py
59:    for chld in dev.signal_names:

FMX/profile_collection/startup/25-zebra.py
17:    return [sig_name for sig_name in cls.signal_names

AMX/profile_collection/startup/25-zebra.py
17:    return [sig_name for sig_name in cls.signal_names

CHX/ipython_ophyd/profile_collection/startup/30-user.py
94:    for k in device.signal_names:

CHX/ipython_ophyd/profile_collection/startup/10-optics.py
22:        for name in self.signal_names:
107:        for name in self.signal_names:
214:        for name in self.signal_names:

LIX/profile_collection/startup/20-detectors.py
204:        for overlay in cam.over1.signal_names:

```

**Comment by danielballan** [comment]
Cool. Just to be sure I'm clear above: this is a deprecation, not an API break.

**Comment by tacaswell** [comment]
attn @ZLLentz @teddyrendahl

**Comment by danielballan** [comment]
Would this generate a ton of annoying warnings at SLAC? If so we can put off the warnings for another release.

**Comment by ZLLentz** [comment]
There is currently only one usage of `signal_names` at SLAC, so don't worry about us. Thanks for the heads-up about the upcoming deprecation.

**Pull request merged**